packages/orm/src/getProperties.ts

```diff
@@ -47,60 +48,78 @@ export function getProperties(meta: EntityMetadata): Record<string, any> {
+  );
+
+  // Enforce transientFields usage
+  const invalidFields = otherFields.filter(([fieldName]) => fieldName !== "transientFields");
+  if (invalidFields.length > 0) {
+    throw new Error(
+      `${meta.type} has invalid class fields, ${invalidFields.map(([k]) => k).join(", ")} should go in transientFields`,
+    );
+  }
+
+  const properties = [
+    // Include the instance-level relations that will be getter-ized by `newEntity`
+    ...relationFields,
+    // And then any prototype-level getters/methods like `isRed` by recursively looking for ownKeys
+    // (this is the previously-mentioned nod to entityResolver to let it copy over getters/methods).
+    ...getRecursivePrototypeKeys(instance)
+      .filter((key) => !knownPrimitives.includes(key))
       .map((key) => {
-        // Return the value of `instance[key]` but wrap it in a try/catch in case it's
-        // a getter that runs code that fails b/c of the dummy state we're in.
         try {
           return [key, (instance as any)[key] ?? unknown];
         } catch {
           return [key, unknown];
         }
-      })
-      // Purposefully return methods, primitives, etc. so that `entityResolver` can add them to the resolver
-      .filter(([key]) => key !== "fullNonReactiveAccess" && key !== "transientFields"),
+      }),
+  ];
+
+  // Keep one version with the relations still lazy, solely for `newEntity`
+  // (technically newEntity will only ask for this once-per-cstr, so a cache is kind of over-kill,
+  // but creating it here, right before we `relationCstr.create`, is a convenient spot).
+  relationConstructors[key] = [...relationFields, ...otherFields];
+
+  // But expose to everyone else the concrete/constructed relations
+  Object.assign(
+    cached,
+    Object.fromEntries(
+      properties.map(([fieldName, value]) => [
+        fieldName,
+        value instanceof RelationConstructor ? value.create(instance, fieldName) : value,
+      ]),
+    ),
   );
 
-  return Object.assign(cached, properties);
+  return cached;
+}
+
+/**
+ * Returns the `RelationConstructor`s (...and transientField) for `meta`.
+ *
+ * Should only be used by `newEntity` while moving relations to the prototype.
+ */
+export function getRelationConstructors(meta: EntityMetadata): [string, RelationConstructor<any> | object][] {
+  getProperties(meta); // We populate the relationConstructors during getProperties
+  const key = meta.stiDiscriminatorValue ? `${meta.tableName}:${meta.stiDiscriminatorValue}` : meta.tableName;
+  return relationConstructors[key];
 }
 
 export class UnknownProperty {}
 const unknown = new UnknownProperty();
 
 const propertiesCache: Record<string, any> = {};
-const fakeInstances: Record<string, Entity> = {};
-
-/**
- * Returns a fake instance of `meta` so that user-defined `CustomReference` and `AsyncProperty`s can
- * be inspected on boot.
- */
-export function getFakeInstance(meta: EntityMetadata): Entity {
```

Comment:
The only caller of `getFakeInstance` really should have been using `getProperties`, so moved them over to that.